scraper/src/youtube2zim/scraper.py

```diff
@@ -1108,3 +1120,240 @@ def to_data_js(video):
+        videos = list(filter(has_channel, videos))
+        for video in videos:
+            slug = get_video_slug(video)
+            self.zim_file.add_item_for(
+                path=f"videos/{slug}.json",
+                title=slug,
+                content=generate_video_object(video).model_dump_json(
+                    by_alias=True, indent=2
+                ),
+                mimetype="application/json",
+                is_front=False,
+            )
+
+        # write playlists JSON files
+        playlist_list = []
+
+        main_playlist_slug = None
+        if len(self.playlists) > 0:
+            main_playlist_slug = get_playlist_slug(
+                self.playlists[0]
+            )  # set first playlist as main playlist
+
+        for playlist in self.playlists:
+            playlist_slug = get_playlist_slug(playlist)
+            playlist_path = f"playlists/{playlist_slug}.json"
+
+            if playlist.playlist_id != self.uploads_playlist_id:
+                playlist_list.append(generate_playlist_preview_object(playlist))
+            else:
+                main_playlist_slug = (
+                    playlist_slug  # set uploads playlist as main playlist
+                )
+
+            self.zim_file.add_item_for(
+                path=playlist_path,
+                title=playlist.title,
+                content=generate_playlist_object(playlist).model_dump_json(
+                    by_alias=True, indent=2
+                ),
+                mimetype="application/json",
+                is_front=False,
+            )
+
+        # write playlists.json file
+        self.zim_file.add_item_for(
+            path="playlists.json",
+            title="Playlists",
+            content=json.dumps(playlist_list, default=pydantic_encoder, indent=2),
+            mimetype="application/json",
+            is_front=False,
+        )
+
+        # write channel.json file
+        self.zim_file.add_item_for(
+            path="channel.json",
+            title=self.title,
+            content=Channel(
+                id=str(self.main_channel_id),
+                title=str(self.title),
+                description=str(self.description),
+                profile_path="profile.jpg",
+                banner_path="banner.jpg",
+                collection_type=self.collection_type,
+                main_playlist=main_playlist_slug,
+            ).model_dump_json(by_alias=True, indent=2),
+            mimetype="application/json",
+            is_front=False,
+        )
+
+        # clean videos left out in videos directory
+        remove_unused_videos(videos)
+
+    def add_files_to_zim(self, root: Path, zim_file: Creator, fpath: Path):
+        """recursively add a path to a zim file
+
+        root:
+            main folder containing all content
+        zim_file:
+            zim Creator
+        fpath:
+            path to the file/folder to add to the ZIM"""
+        if fpath.is_dir():
```

Comment:
Maybe add something like `logger.debug("Adding files to ZIM")` so that next lines are clearer